scripts/intfutil

```diff
@@ -221,12 +223,16 @@ def port_optics_get(db, intf_name, type):
     Get optic type info for port
     """
     full_table_id = PORT_TRANSCEIVER_TABLE_PREFIX + intf_name
+    port_id = PORT_TABLE_PREFIX + intf_name
+    port_role = db.get(db.CONFIG_DB, port_id, PORT_ROLE)
     optics_type = db.get(db.STATE_DB, full_table_id, type)
     if optics_type is None:
         if is_rj45_port(intf_name):
             return OPTICS_TYPE_RJ45
         elif db.get(db.APPL_DB, PORT_STATUS_TABLE_PREFIX + intf_name, multi_asic.PORT_ROLE) == multi_asic.DPU_CONNECT_PORT:
             return TYPE_DPC
+        elif port_role == TYPE_DPC:
```

Comment:
we already have the logic to print DPU port type output in the cli. 
it's right above this line:

```
elif db.get(db.APPL_DB, PORT_STATUS_TABLE_PREFIX + intf_name, multi_asic.PORT_ROLE) == multi_asic.DPU_CONNECT_PORT:
            return TYPE_DPC
```

Why do we need this extra line?

Reply:
For SmartSwitch platforms the condition above will not be true

Reply:
Can you explain why is it not? role field is populated in the APPL_DB and intfutil output shows the correct output

```
redis-cli -n 0 hget "PORT_TABLE:Ethernet232" role
"Dpc"


show int status
    Ethernet224  224,225,226,227,228,229,230,231     400G   9100    N/A    etp29           routed    down     down                                DPU-NPU Data Port         off
    Ethernet232  232,233,234,235,236,237,238,239     400G   9100    N/A    etp30           routed    down     down                                DPU-NPU Data Port         off
    Ethernet240  240,241,242,243,244,245,246,247     400G   9100    N/A    etp31           routed    down     down                                DPU-NPU Data Port         off
    Ethernet248  248,249,250,251,252,253,254,255     400G   9100    N/A    etp32           routed    down     down                                DPU-NPU Data Port         off
```

Reply:
@vivekrnv The line "elif db.get(db.APPL_DB, PORT_STATUS_TABLE_PREFIX + intf_name, multi_asic.PORT_ROLE) == multi_asic.DPU_CONNECT_PORT:" was evaluating to be "False" in the past.  I tested it now and is "True" now.  So, the next line can be removed.

Reply:
All the changes to intfutil can be removed, i think

Reply:
@vivekrnv everything besides the   define TYPE_DPC = 'Dpc'   can be removed.
The  TYPE_DPC = 'Dpc' was captured in the PMON HLD.
Can you confirm that TYPE_DPC = 'DPU-NPU Data Port' vs TYPE_DPC = 'Dpc' ?

Reply:
@vivekrnv Will close it for now with "TYPE_DPC = 'DPU-NPU Data Port'" and if anyone raises concern then we can revisit.